locopy/database.py

```diff
@@ -93,29 +65,14 @@ class Base(object):
         Database credentials are not provided or valid
     """
 
-    def __init__(
-        self,
-        dbapi=None,
-        host=None,
-        port=None,
-        database=None,
-        user=None,
-        password=None,
-        config_yaml=None,
-        **kwargs
-    ):
+    def __init__(self, dbapi, config_yaml=None, **kwargs):
         self.dbapi = dbapi
-        self.host = host
-        self.port = port
-        self.database = database
-        self.user = user
-        self.password = password
-        self.extra_conn = kwargs or {}
+        self.connection = kwargs or {}
         self.conn = None
         self.cursor = None
 
         if config_yaml:
-            self.__dict__.update(get_redshift_yaml(config_yaml))
+            self.connection = read_config_yaml(config_yaml)
```

Comment:
May want to either exception out if someone provides both config_yaml and kwargs or use kwargs to update whatever's read from config_yaml?

Reply:
Right now it will just overwrite with whatever is in the YAML. Maybe a warning? Or something along the lines of:

```python
if config_yaml and not self.connection:
    ...
else:
    Exception(...
```